### DISCUSSION TITLE
Kotlin - IntelliJ IDE cannot reslove the references from synthetic class

### DISCUSSION THREAD

**Discussion by jacobzhang1979** [arrow-kt/arrow#2629] [General]
**Problems Description:**
I have the kotlinc compiler plugin (using arrow-meta library) in place which changes the .class by adding for example new properties or new method etc during kotlin compilation time. for example, the orginal source Kotlin A.kt is like below

@MetaData
data class A (val x: String, val y: String)

after applying compiler plugin, the .class will be altered to (from source perspective), basically I will add implicitly MetaData into the primary constructor for all class so long as it is with annonation **@MetaData** in place, plus a new method _fun getMetaData()_  generated

data class A(val x: String, val y:String, val myMeta: MetaData) {
   fun getMetaData() = {
       //logic check
       return myMeta
   }
}

now when it comes to use the new "class" manipulated as below, IntelliJ complains it cannot find resolve A (it has only the constructor with 2 parameters not 3) and cannot resolve the the synthetic method _getMetaData()_ either.
val x = A("ba", "fo", MetaData(..))
val y = x.getMetaData()

can somebody shed some light on it?  

I know lombok seems no problem with it after adding its @Getter annotation for example into Java source code, IntelliJ can recognize its getXXX method (which is generated by lombok). 
I don't know how to implement the same for my case for kotlin language. please kindly detail the steps if possible.

Best Regards
Jianbing

**Comment by kyay10** [comment]
This is expected behaviour because of how Arrow Meta operates. Regardless, Arrow Meta is K1 only, so this is outdated
